src/sql/azdata.proposed.d.ts

```diff
@@ -1790,6 +1795,75 @@ declare module 'azdata' {
 		 * The path of the parent node.
 		 */
 		parentNodePath: string;
+		/**
+		 * list of filters that are supported by the node
+		 */
+		filterableProperties?: NodeFilterProperty[];
+	}
+
+	export interface NodeFilterProperty {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeFilterPropertyDataType;
+		/**
+		 * The description of the filter property
+		 */
+		description: string;
+	}
+
+	/**
+	 * NodeFilterChoiceProperty is used to define the choices for the filter property if the type is choice
+	 */
+	export interface NodeFilterChoiceProperty extends NodeFilterProperty {
+		/**
+		 * The list of choices for the filter property if the type is choice
+		 */
+		choices: string[];
+	}
+
+	export interface NodeFilter {
+		/**
+		 * The name of the filter property
+		 */
+		name: string;
+		/**
+		 * The type of the filter property
+		 */
+		type: NodeFilterPropertyDataType;
```

Comment:
why do we need this here, isn't this implied by the name of the property?

Reply:
It's just the datatype of the property. For ex: Name is string, create date is date

Reply:
that seems to be redundant, you should look for the type by the `name` property.

Reply:
Not sure what you mean, how would the ADS side know what names map to what types? These are going to coming from providers so could be anything.

Reply:
Oh wait, sorry. Thought this was in the other place - you're right Alan this seems unnecessary. The provider is the one getting this and so should know what names map to what types.